File: frontend-web/publisher-app/src/app/shared/models/Review.ts
export class Review {
    postId: number;
    approved: boolean;
    reviewMessage: string | null | undefined;

    constructor(postId: number, approved: boolean, reviewMessage: string) {
        this.postId = postId;
        this.approved = approved;
        this.reviewMessage = reviewMessage;
    }
}

File: backend-java/SpringCloudPXL/review-service/src/main/java/be/pxl/services/domain/Review.java
package be.pxl.services.domain;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.time.LocalDateTime;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Review implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    private long postId;
    private String reviewMessage;
    private boolean approved = false;
    private LocalDateTime creationDate;

}
